compiler/src/dotty/tools/dotc/semanticdb/DiagnosticOps.scala

```diff
@@ -0,0 +1,22 @@
+package dotty.tools.dotc.semanticdb
+
+import dotty.tools.dotc.reporting.Diagnostic
+import dotty.tools.dotc.{semanticdb => s}
+import dotty.tools.dotc.interfaces.Diagnostic.{ERROR, INFO, WARNING}
+import dotty.tools.dotc.core.Contexts.Context
+
+object DiagnosticOps:
+  private val asciiColorCodes = "\u001B\\[[;\\d]*m".r
+  extension (d: Diagnostic)
+    def toSemanticDiagnostic(using Context): s.Diagnostic =
+      val severity = d.level match
+        case ERROR => s.Diagnostic.Severity.ERROR
+        case WARNING => s.Diagnostic.Severity.WARNING
+        case INFO => s.Diagnostic.Severity.INFORMATION
+        case _ => s.Diagnostic.Severity.INFORMATION
+      val msg = asciiColorCodes.replaceAllIn(d.msg.message, m => "")
+      s.Diagnostic(
+        range = Scala3.range(d.pos.span, d.pos.source),
```

Comment:
actually it seems like `Scala3.range` can drop its context parameter and so can `toSemanticDiagnostic` so then you can call it in parallel

Reply:
actually now maybe still not safe as `isVerbose` in the diagnostic is mutable